libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -0,0 +1,432 @@
+        let request = self.requestForDelete(withObjectType: objectType, objectId: objectId, apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a create.
+    /// - Parameter objectType: An object type.
+    /// - Parameter fields: Dictionary of fields
+    /// - Parameter completionBlock: The completion block to invoke.
+    func create(_ objectType: String, fields: [String:Any], _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.requestForCreate(withObjectType: objectType, fields: fields, apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute resources request
+    /// - Parameter completionBlock: The completion block to invoke.
+    func resources(_ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forResources: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute versions request
+    /// - Parameter completionBlock: The completion block to invoke.
+    func apiVersions(_ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.requestForVersions()
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a searchScopeAndOrder request
+    /// - Parameter completionBlock: The completion block to invoke.
+    func searchScopeAndOrder(_ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forSearchScopeAndOrder: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a searchResultLayout
+    /// - Parameter objectList: Array of objects.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func searchResultLayout(_ objectList:[String], _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forSearchResultLayout: objectList.joined(separator: ","), apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+     
+    /// Execute a Composite request
+    /// - Parameter requests: Array of RestRequewst
+    /// - Parameter referenceIds: String array of reference identifiers.
+    /// - Parameter allOrNone: Boolean to indicate whether to set allOrNone.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func composite(_ requests:[RestRequest], referenceIds: [String], allOrNone: Bool, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
```

Comment:
We could do better than a generic RestResponse.
The response body contains an array with the responses for each sub request. For more information see: https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/responses_composite.htm